golem-worker-service/src/api/openapi_generator.rs

```diff
@@ -0,0 +1,437 @@
+// Copyright 2024 Golem Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use golem_api_grpc::proto::golem::apidefinition::{HttpApiDefinition, HttpRoute};
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use std::collections::HashMap;
+use poem_openapi::Object;
+
+/// Represents an OpenAPI Specification document.
+#[derive(Serialize, Deserialize, Object)]
+pub struct OpenApiSpec {
```

Comment:
It could be useful to use a third-party crate that already models OpenAPI 3.0 (I found two with a quick search).